src/components/menu/item-menu.tsx

```diff
@@ -1,26 +1,39 @@
 import { colors } from '../../typography/colors';
 import { Label } from '../typography/label/label';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import './menu.css';
+import { useLocation, Link } from 'react-router-dom';
+
 interface ItemProps {
   title: string;
   redirect: string;
 }
 
 export function ItemMenu({ title, redirect }: ItemProps) {
+  const [isCurrentRoute, setCurrentRoute] = useState(false);
   const [hover, setHover] = useState(false);
+  const location = useLocation();
+
+  useEffect(() => {
+    if (location.pathname === redirect) {
+      setCurrentRoute(true);
+    } else {
+      setCurrentRoute(false);
+    }
+  }, [location.pathname]);
```

Comment:
geralmente a sicronizacao de estados a partir de alguma prop não é necessitado o uso de um useEffect, e nesse caso em especifico, como apenas o useEffect que atualiza o valor do estado, vc pode jogar fora ambos o estado e o useEffect, podendo declarar a constante "inline":

```suggestion
  const isCurrentRoute = location.pathname === redirect;
```